### DISCUSSION TITLE
[Proposal] Bees

### DISCUSSION THREAD

**Discussion by drumdrumdrum3337** [sunflower-land/sunflower-land#2230] [Ideas]
Real-world bees maintenance and honey production are quite complex and we probably wouldn't to want to transfer it all directly into the game, however some of that complexity could bring new game loops, and this proposal tries to explore how to do just that.

**Basic premises:**
1. Bees need a beehive to live in. It is a wooden structure that is built once and can serve many bee families over the years
2. Bees need flowers to harvest pollen and nectar in order to feed their colony and produce honey and beeswax
3. Each bees colony, if there's enough food, eventually creates a new colony that needs a separate beehive to live in
4. Bees colonies eventually die, they have an expiration time
5. If there are not enough flowering plants around, then colonies will not produce honey and beeswax. If there's a total lack of flowers, then colonies will eventually die
6. Bees are pollinators beneficial for the plants they pollinate
7. Honey and beeswax production cannot add-up with every new beehive added - the number of flowering plants within the reach of bees is always limited, hence there's always a limit of beehives that makes sense to build.

**Mechanics**

**1. Beehive**
This is a simple structure that requires a decent amount of wood (e.g. 500), some iron (e.g. 20) and mastery to craft (e.g. 150sfl).

**2. Starting a bees colony**
Once beehive is built, farmer has to purchase first bee colony from the shop. Depending on how expanded the farm is and how many flowering plants it could hold (more on this below), the farmer could buy more than one colony and build more than one beehive right from the start. However, colonies are pricey (e.g. 150sfl) and the new farmer should decide whether it is fine for the farmer to invest in all the new colonies right from the start, or to start with a lesser number of colonies, wait for them to develop, create new colonies and occupy new beehives. 
Note: potentially, farmer could be allowed to build more beehives to host colonies but the honey production per beehive should drop.

**3. Feeding bees**
Bees can feed themselves from two sources: 'active' fruit trees and bushes that are already on the farm, and specifically planted/placed annual and perennial flowers. 

Feeding from fruit trees and bushes would be enough for the current colonies to survive till their 'expiration', but it won't be enough for the development of new colonies and honey and beeswax production. Note: by pollinating fruit trees and bushes, bees increase harvest of fruits and berries (e.g. +20%). 'Active' fruit trees and berries are not required for honey and beeswax production.

For honey and beeswax production, bees need plenty of specifically planted flowers. To avoid occupying veggies beds and the necessity to create new type of beds, the suggestion is to use potted plants already available in the Decoration shop, and expand the assortment to include annual (cheap, but die faster) and perennial (more expensive, but live longer (not infinite though)) plants. The number of potted flowers available per farm should depend on the number of current expansions, since the space is limited. The number of available plants to place determine the number of possible beehives to be built.

If there are no 'active' fruit trees and berry bushes and no potted plants, bees colonies will die quite quickly (in a matter of days).

**4. Harvesting honey**
Thus far, everything that farmer did was building beehive(s), buying colony(ies) and placing potted plants. After that, bees get to work. In some time (e.g. in 1 week) they start producing honey, e.g. 1 honey from each beehive every 12hrs. Farmer just clicks on the beehive to collect honey once it is ready.
Note: describing use cases for honey is out of scope for this proposal.

**5. Harvesting beeswax**
Once per lifetime of each colony, farmer can gather beeswax, and sell it in the Decorations shop. Use cases for wax need to be developed, but it looks like it is not used much outside of cosmetics and candles (hence selling it in the decoration shop). 

**6. New colony**
Twice in its lifetime, each colony produces a new colony. It could be transferred to a new beehive on the farm, or sold in the shop for a decent amount of sfl (e.g. if buying new colony costs 150sfl, then selling the new one to the shop could be around 100sfl). New colony cannot be 'stored' for long, it needs to be either sold to the shop or moved to an available beehive within 48 hours, otherwise it heads away into the wild and is lost to the farmer. 

**7. Example of a bees colony lifecycle**
Once new bees colony inhabits a beehive, the timer for its 'expiration' starts. Every third of its lifecycle, the colony produces a new colony that needs a new home. At the end of the lifecycle, colony dies, and the beehive becomes vacant.

For example, colony (A) lives for 4.5 months. After 1.5 months, colony (A) produces a new colony (B). 

Let's say, the farmer has put colony (B) into an available beehive. After another 1.5 months, the colony (A) produces a colony (C) and colony (B) produces new colony (D). The farmer sells both (C) and (D) to the shop. After another 1.5 months, colony (A) dies and the beehive becomes vacant. This happens roughly at a time when the colony (B) produces new colony (E). 

If farmer is attentive, the colony (E) can occupy the beehive vacated by the colony (A). 

This means that starting with 1 colony, after 3 months the farmer gets 2 colonies occupying 2 beehives, while two other colonies were sold to the shop.

**Summary**
This is indeed a high level 'bare' proposal in a sense that there's very little actual calculations involved, as bees, afaik, are still quite far away on the roadmap and the actual calculations should depend on the state of the economy, bumpkins levels and average expansions at the time of launch. 

There are also no concepts such as bees' rarity, breeds, or predators described, as the goal was to describe the most basic gameplay for bees, everything else could be added later once there's an agreement on the very basics.

Looking forward to your comments!

**Comment by Tambel1979** [comment]
I would love to see bees in Sunflower Land. Realistically, crops won't grow without them. Fruit trees need them, especially apples and oranges, so do berry bushes like blueberries. They could even be a boost for crops that they pollinate. Not now, but later on in the game, there might be a use for beeswax in potions and lotions and candles, if we ever have crafters of such things, which would be nice. Our bumpkins might like some lip balm or a lotion after being out on the farm all day.

At any rate, I'm all in on bees, because no farm can exist without them. And I'm also all in on flowers and ornamentals, which would go along with bees.
